detectors/gradle/direntry/direntry.go

```diff
@@ -0,0 +1,156 @@
+package direntry
+
+import (
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+)
+
+var ignoreDirs = []string{".git", ".github", ".gradle", ".idea", "build", ".kotlin", ".fleet", "CordovaLib", "node_modules"}
+
+type DirEntry struct {
+	AbsPath string
+	RelPath string
+	Name    string
+	IsDir   bool
+	entries []DirEntry
+}
+
+// WalkDir walks the directory tree starting from rootDir and returns a DirEntry representing the root directory.
+func WalkDir(rootDir string, depth uint) (*DirEntry, error) {
+	if depth == 0 {
+		return nil, nil
+	}
+
+	parent := DirEntry{
+		AbsPath: rootDir,
+		RelPath: "./",
+		Name:    "",
+		IsDir:   true,
+		entries: nil,
+	}
+
+	if err := recursiveWalkDir(rootDir, &parent, 0, depth); err != nil {
+		return nil, err
+	}
+
+	return &parent, nil
+}
+
+// FindFirstEntryByName returns the first entry (shortest file path) with the specified name and directory status.
+func (e DirEntry) FindFirstEntryByName(name string, isDir bool) *DirEntry {
+	return recursiveFindFirstEntryByName([]DirEntry{e}, name, isDir)
+}
+
+// FindDirectEntryByName returns the direct child entry with the specified name and directory status.
+func (e DirEntry) FindDirectEntryByName(name string, isDir bool) *DirEntry {
```

Comment:
nit: I had to read the comment to fully understand what `FindDirectEntryByName` does. It is clear that it only tries to find the entry in the first child level. I do not have a good name suggestion yet but maybe `FindDirectChildEntryByName` could be better. I am not a 100% sure in this name so this comment is totally optional.